# Do not show an error dialog when not running under vbox
# Do not start VBoxClient --vmsvga, we run VBoxClient --vmsvga as
# a systemd service, this works with both Wayland and Xorg based sessions

---
 src/VBox/Additions/x11/Installer/98vboxadd-xclient |   13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient VirtualBox-7.0.2/src/VBox/Additions/x11/Installer/98vboxadd-xclient
--- VirtualBox-7.0.2.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient	2022-10-19 21:20:45.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Additions/x11/Installer/98vboxadd-xclient	2022-10-22 12:05:28.770187860 +0300
@@ -31,18 +31,11 @@ for i in $HOME/.vboxclient-*.pid; do
     test -w $i || rm -f $i
 done
 
-if ! test -c /dev/vboxguest 2>/dev/null; then
-   # Do not start if the kernel module is not present.
-   # Execute notify-send in the back-ground to avoid racing with sddm,
-   # as notify-send may wait for sddm to start while it waits for us to exit.
-   notify-send "VBoxClient: the VirtualBox kernel service is not running.  Exiting." &
-elif test -z "${SSH_CONNECTION}"; then
-   # This script can also be triggered by a connection over SSH, which is not
-   # what we had in mind, so we do not start VBoxClient in that case.  We do
-   # not use "exit" here as this script is "source"d, not executed.
+# Do not start if the kernel module is not present; or if this script is
+# triggered by a connection over SSH.
+if [ -c /dev/vboxguest -a -z "${SSH_CONNECTION}" ]; then
   /usr/bin/VBoxClient --clipboard
   /usr/bin/VBoxClient --checkhostversion
   /usr/bin/VBoxClient --seamless
   /usr/bin/VBoxClient --draganddrop
-  /usr/bin/VBoxClient --vmsvga-session # In case VMSVGA emulation is enabled
 fi

